languages/elm/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,119 @@
 # How to implement an Elm concept exercise
 
-TODO: describe how to implement a concept exercise for the Elm track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Elm track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+Any concept exercise in any v3 track requires the following files to be created:
+
+<pre>
+languages
+└── elm
+    └── exercises
+        └── concept
+            └── &lt;SLUG&gt;
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                └── .meta
+                |   └── config.json
+                |   └── design.md
+                |   └── Example.elm
+                ├── elm.json
+                ├── src
+                |   └── &lt;NAME&gt;.elm
+                └── tests
+                |   └── &lt;NAME&gt;.elm
+</pre>
+
+## Step 1: add .docs/introduction.md
+
+This file contains an introduction to the concept. It should be explicit about what the exercise teaches and maybe provide a brief introduction to the concepts, but not give away so much that the user doesn't have to do any work to solve the exercise.
+
+## Step 2: add .docs/instructions.md
+
+This file contains instructions for the exercise. It should explicitly explain what the user needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+
+## Step 3: add .docs/hints.md
+
+If the user gets stuck, we will allow them to click a button requesting a hint, which shows this file. We will softly discourage them using it. The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student.
```

Comment:
We recently updated the text:

```suggestion
If the student gets stuck, we will allow them to click a button requesting a hint, which shows this file. This will not be a "recommended" path and we will (softly) discourage them using it unless they can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated.

The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used.

The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
```